Singularity/Singularity/Platform/PlatformBlank.cs

```diff
@@ -132,34 +139,35 @@ public void TakeHealDamage(int damage)
         /// Add a new resource to the platform.
         /// </summary>
         /// <param name="resource"> the resource to be added to the platform </param>
-        public void StoreResource(IResource resource)
+        public void StoreResource(Resource resource)
         {
             mResources.Add(resource);
         }
 
         /// <summary>
         /// Use this method to get the resource you asked for. Removes the resource from the platform.
         /// </summary>
-        /// <param name="resource">The resource you ask for</param>
+        /// <param name="resourcetype">The resource you ask for</param>
         /// <returns>the resource you asked for, null otherwise.</returns>
-        public IResource GetResource(IResource resource)
+        public Resource GetResource(EResourceType resourcetype)
         {
-            var index = mResources.IndexOf(resource);
-            if (index < 0)
-            {
-                return null;
-            }
+            // var index = mResources.FindIndex(x => x.isType(resourcetype));
```

Comment:
It looks good, why is this commented?

Reply:
Because something did not compile at the time of writing - I'll look over it again